magellan-library/src/main/java/com/wealthfront/magellan/navigation/CurrentNavigableProvider.kt

```diff
@@ -0,0 +1,16 @@
+package com.wealthfront.magellan.navigation
+
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class CurrentNavigableProvider @Inject constructor() : NavigableListener {
+
+  lateinit var navigable: NavigableCompat
```

Comment:
Should probably add a `private set` to this

Reply:
yeah. can do.